### ISSUE TITLE
Reference strip using "thumbs"

### ISSUE THREAD

**Issue by VipSaran** [openseadragon/openseadragon#1209]
I've had some (big) performance issues with my current setup (I'm using OSD for timelapse viewer) as I have up to 60 images for the same day which I load for ReferenceStrip. Problems are rooted in image sizes (those being 1-2 MB, or in original quality 5-8 MB). This automatically breaks the page on many mobile/touch devices and causes slow (and unnecessary) loading otherwise.

I've, in porcess of trying to solve that, uncovered and "solved" the  #1172 , but nevertheless I had to disable the RS for the performance issues it was causing.

During my search for the solution (which I imagined would be to just provide "thumbs" for RS to load instead), I have stumbled upon this issue: #1045 , but the approaches there do not satisfy me! :)

What I have in mind (and what I have actually implemented) is a much simpler solution reusing and extending existing Viewer/RS functionality.

I've taken this basic initialization:
```
OpenSeadragon({
    ...
    tileSources: {
        type: 'image',
        url:  '/example-images/grand-canyon-landscape-overlooking.jpg'
    }
    ...
});
```
and just extended it with `thumb_url`:
```
OpenSeadragon({
    ...
    tileSources: {
        type: 'image',
        url:  '/example-images/grand-canyon-landscape-overlooking.jpg'
        thumb_url:  '/example-images/thumbs/grand-canyon-landscape-overlooking.jpg'
    }
    ...
});
```
Next, in ReferenceStrip's `loadPanels`, I'm cloning original tileSources and, if they have the `thumb_url` specified, I'm using its value for the miniViewer.

Something (exactly) like this:
```
        if ( !element.activePanel ) {
            // clone to preserve the URL of the original (large) image for the tileSource
            // as Viewer references it
            var miniTileSource = $.extend(true, {}, strip.viewer.tileSources[i]);

            if (miniTileSource.thumb_url !== undefined) {
              miniTileSource.url = miniTileSource.thumb_url;
            }

            miniViewer = new $.Viewer( {
                id:                     element.id,
                tileSources:            [miniTileSource],
                ...
```
I know it adds some overhead (time and memory), but I believe those are (greatly) compensated by same benefits from using smaller images (thumbs) for the reference strip.

NB. I'm only using single images (non-tiled), so I don't know if this somehow affects the actual tiled images in RS.

NB2. Also, apart from `getImageRecord()` function, I don't see any other sanity/type/existance check, so my check for `thumb_url` being defined should suffice. Or?

Does this make sense or have I missed some pattern how same could be achieved with current Viewer/RS implementation?

**Comment by iangilman** [comment]
That seems like a great solution! A couple refinements: 

* I think we should be explicit about what the thumbnail url is for, so it should be something like `referenceStripThumbnailUrl` instead of `thumb_url`, since it has that one very specific function. Also having a long name like that it's less likely to collide with properties that might already exist in other tile sources. 
* The code as written won't work so well if your tile source isn't `image` but I do think for explicit thumbnail urls it should always be `image`. That can be accomplished something like so: 

```
if ( !element.activePanel ) {
    var miniTileSource;
    var originalTileSource = strip.viewer.tileSources[i];
    if (originalTileSource.referenceStripThumbnailUrl) {
        miniTileSource = {
            type: 'image',
            url: originalTileSource.referenceStripThumbnailUrl
        };
    } else {
        // clone to preserve the URL of the original (large) image for the tileSource
        // as Viewer references it
        miniTileSource = $.extend(true, {}, originalTileSource);        
    }

    miniViewer = new $.Viewer( {
        id:                     element.id,
        tileSources:            [miniTileSource],
        ...
```

Would you be up for making a patch along the lines you describe?

**Comment by iangilman** [comment]
Fixed by #1241
